node/malus/src/variants/common.rs

```diff
@@ -236,28 +243,91 @@ where
 								),
 							})
 						}
-						create_validation_response(
-							validation_data,
-							candidate_receipt.descriptor,
-							sender,
-						);
-						None
+						// Create the fake response with probability `p` if the `PoV` is malicious.
+						let behave_maliciously = self.distribution.sample(&mut rand::thread_rng());
+						match behave_maliciously {
+							true => {
+								gum::info!(
+									target: MALUS,
+									"😈 Malicious behaviour is sampled as: {:?}. Now outputting malicious ValidationResult::Valid message.",
+									&behave_maliciously,
+								);
+
+								create_validation_response(
+									validation_data,
+									candidate_receipt.descriptor,
+									sender,
+								);
+								None
+							},
+							false => {
+								// Behave normally with probability `(1-p)` for a malicious `PoV`.
+								gum::info!(
+									target: MALUS,
+									"😈 Malicious behaviour is sampled as: {:?}. Faking normal behaviour and outputting CandidateValidationMessage::ValidateFromExhaustive.",
```

Comment:
This is actually not faking anything, just passing the message to the real candidate validation subsystem.